doc/languages-frameworks/perl.section.md

```diff
@@ -0,0 +1,179 @@
+# Perl {#sec-language-perl}
+
+## Running perl programs on the shell {#ssec-perl-running}
+
+When executing a Perl script, it is possible you get an error such as `./myscript.pl: bad interpreter: /usr/bin/perl: no such file or directory`. 
+This happens when the script expects Perl to be installed at `/usr/bin/perl`, which is not the case when using Perl from nixpkgs. 
+You can fix the script by changing the first line to:
+
+```ShellSession
+#!/usr/bin/env perl
+```
+
+to take the Perl installation from the `PATH` environment variable, or invoke Perl directly with:
+
+```ShellSession
+$ perl ./myscript.pl
+```
+
+When the script is using a Perl library that is not installed globally, you might get an error such as `Can't locate DB_File.pm in @INC 
+(you may need to install the DB_File module)`. In that case, you can use `nix-shell` to start an ad-hoc shell with that library installed, for instance:
+
+```ShellSession
+$ nix-shell -p perl perlPackages.DBFile --run ./myscript.pl
+```
+
+If you are always using the script in places where `nix-shell` is available, you can embed the `nix-shell` invocation in the shebang like this:
+
+```ShellSession
+#!/usr/bin/env nix-shell
+#! nix-shell -i perl -p perl perlPackages.DBFile
+```
+
+## Packaging Perl programs {#ssec-perl-packaging}
+
+Nixpkgs provides a function `buildPerlPackage`, a generic package builder function for any Perl package that has a standard `Makefile.PL`. It’s implemented in 
```

Comment:
I wasn't aware that there is a formatter. I will try it out next.

Reply:
@jtolnar I am unable to squash my commits. Trying to follow the instructions here https://medium.com/@mittalyashu/how-to-squash-commits-in-a-github-pull-request-97b36576eacb leads to a tremendous amount of merge conflicts by commits that I didn't make... Please advise.
"Auto-merging pkgs/top-level/all-packages.nix
CONFLICT (content): Merge conflict in pkgs/top-level/all-packages.nix
error: could not apply a64d7d84456... Revert "Revert "texinfo: revert to version 6.5 on Darwin""
"

Reply:
For squashing, I would suggest rebasing onto the parent commit of your first commit instead of `master`. That way you will have less commits to worry about.

```
git rebase -i c7a82285f6c63816b66e78fa1b66fb2bfbe67f8f^
```

But note that you will need to rebase onto `master` anyway, to resolve the conflict in `index.xml`.